### PR TITLE
Feature/frontend claim

### PR DESCRIPTION
**Pull request by jordaniza**
Adds claim functionality to the frontend. 

When a bounty is created using the frontend react application, we are missing certain elements compared to the situation in which the bounty is created from discord, namely a discord message id to point the user.

This functionality creates a new flow from the website. It is intended to be coupled with an event listener setup on the discord bot, whereby a user can claim directly within the web app. Upon claiming, the status of the bounty is moved from 'Open' to 'In-Progress', the claimant is the discord user currently logged in, and the status history is updated.

A message is added to 'submissionNotes', which can be forwarded on to the discord user who created the bounty.

There are now 2 claim workflows:

- If we have a discordmessageId, we will use it: the user will be taken to the bounty in discord
- If we don't (the bounty was created with the web ui), then the user can 'claim' using the frontend flow.
- This will mark the bounty as claimed in the UI, but, importantly, until we have a linkage to the creator on discord (or the web UI) then the creator will not receive a notification.

Added tests on the backend and some basic frontend tests as well. This release also adds a few stability improvements:

- The BBBS server is whitelisted for create and claim
- Users with the `admin` role have full access to all board functionality
- Fixed formatting with the DAO Selector

### CODE REVIEW

File: packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx:42]: Own Review: no handler here for errors

File: packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx:57]: discordMessageId is a required field, the create form stores as `''` in the web ui which coalesces to `false`. 

It would probably be better to have `discordMessageId` be an optional field instead. Same result but much more declarative

File: packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx
Comment (jordaniza) [inline packages/react-app/src/components/pages/Bounties/Bounty/claim.tsx:116]: extracted hook to make it easier to test - mocking the useContext object with Chakra creates all sorts of headaches

File: packages/react-app/src/pages/api/bounties/[id].ts
Comment (jordaniza) [inline packages/react-app/src/pages/api/bounties/[id].ts:18]: editKey was legacy schema, we use RBAC for access control instead so I have removed references to it

File: packages/react-app/src/pages/api/bounties/[id]/claim.ts
Comment (jordaniza) [inline packages/react-app/src/pages/api/bounties/[id]/claim.ts:8]: I set claim as a separate route because I wanted to segment claimants from editors. I see these as different permissions.

File: packages/react-app/src/services/auth.service.ts
Comment (jordaniza) [inline packages/react-app/src/services/auth.service.ts:85]: I moved this here to make it much more obvious that permissions cached should wrap getPermissions. At the top of the file it's easy for a developer to assume that getPermissions should be used, and they will run into discord api limits

File: packages/react-app/src/services/auth.service.ts
Comment (jordaniza) [inline packages/react-app/src/services/auth.service.ts:95]: When we move to multitenant, this will just be a lookup from the applicable roles or something similar

File: packages/react-app/src/utils/dbConnect.ts
Comment (jordaniza) [inline packages/react-app/src/utils/dbConnect.ts:39]: fixes buffer unhandled error

**Review by Behold3th — DISMISSED** [review]
Didn't have a chance to go through line by line, but the preview build looked beautiful

approved, but if you could get that `statusHistory` change in to track the open state before you merge, that would be fantastic 

🔥 work as always, @jordaniza

**Pull request merged**
